src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -7,6 +7,20 @@ import { SavedObjectAttributes } from 'src/core/types';
 
 export interface DataSourceAttributes extends SavedObjectAttributes {
   title: string;
+  description?: string;
   endpoint: string;
   noAuth: boolean;
+  credentials?: {
```

Comment:
Since There can be only 1 credential, why name is credentials?

Reply:
> Since There can be only 1 credential, why name is credentials?

From @dblock - "I believe for credentials it's always "Credentials", plural. A credential is like accreditation."